src/applications/gi/actions/index.js

```diff
@@ -199,3 +203,46 @@ export function calculatorInputChange({ field, value }) {
 export function toggleFilter() {
   return { type: FILTER_TOGGLED };
 }
+
+const beneficiaryZIPRegExTester = /\b\d{5}\b/;
+
+export function beneficiaryZIPCodeChanged(beneficiaryZIP) {
+  // pass input through to reducers if not five digits
+  if (!beneficiaryZIPRegExTester.exec(beneficiaryZIP)) {
+    return {
+      type: BENEFICIARY_ZIP_CODE_CHANGED,
+      beneficiaryZIP
+    };
+  }
+
+  const url = `${api.url}/zipcode_rates/${beneficiaryZIP}`;
+
+  return dispatch => {
+    fetch(url, api.settings)
+      .then(res => res.json())
+      .then(payload => {
+        if (payload.errors) {
+          dispatch({
```

Comment:
This doesn't follow the pattern of the other actions in this file but I couldn't get the error callback to fire when I received a 404 from this end point. Had to check for an errors attribute to get it to work.

Reply:
Doesn't a 404 put you into the `catch` block? I'm not sure I understand what you mean here.

Reply:
Yeah I don't get it.

![screen shot 2018-06-28 at 10 53 32](https://user-images.githubusercontent.com/4998130/42048797-8844d776-7ac1-11e8-9742-9ed4ce84f496.png)
![screen shot 2018-06-28 at 10 53 20](https://user-images.githubusercontent.com/4998130/42048802-8986269e-7ac1-11e8-8a64-8d1be8b4d042.png)

Reply:
Might be something in `api.settings`?

Reply:
Should also note that I don't think this app  handles any error condition- the fetches have error catches but nothing in the _behavior_ recovers from them. an invalid institution will cause the app to hang :/

Reply:
Alrighty just confirmed that the catch is just for network errors. To handle the 404 I need to move further up the chain and check for `res.ok` (duh) and throw an error if it's not